### PR TITLE
more support for 2D texture arrays in the driver

### PR DESCRIPTION
**Pull request by pixelflinger**
Very much untested, but it should work!

### CODE REVIEW

File: filament/src/driver/opengl/OpenGLDriver.cpp [hunk 1]
```diff
@@ -952,7 +952,7 @@ void OpenGLDriver::textureStorage(OpenGLDriver::GLTexture* t,
             glTexStorage2D(t->gl.target, GLsizei(t->levels), t->gl.internalFormat,
                     GLsizei(width), GLsizei(height));
             break;
-        case GL_TEXTURE_3D: {
```
Comment (romainguy) [inline filament/src/driver/opengl/OpenGLDriver.cpp]: Why the change here?

Comment (romainguy) [reply filament/src/driver/opengl/OpenGLDriver.cpp]: The docs say that both are valid but since the API is called "3D" the "3D" constant makes more sense.

Comment (romainguy) [reply filament/src/driver/opengl/OpenGLDriver.cpp]: Apparently they are different though, we might want to support both. 3D textures let you interpolate across slices, which is useful for color LUTs for instance. 2D arrays don't support that. Same for mipmaps, they behave differently.

Comment (pixelflinger) [reply filament/src/driver/opengl/OpenGLDriver.cpp]: Yeah -- for now I want to support only 2D arrays. We can add 3D textures later, but that will require more API changes at the high level. Like the introduction of a SAMPLER_3D constant. It will be easy though.
